pkg/client/client.go

```diff
@@ -269,10 +269,14 @@ func (c *client) startCheckin() {
 				var lastSentMessage string
 				var lastSentPayload string
 				for {
+					t := time.NewTimer(500 * time.Millisecond)
 					select {
 					case <-done:
+						if !t.Stop() {
+							<-t.C
+						}
 						return
-					case <-time.After(500 * time.Millisecond):
+					case <-t.C:
 					}
```

Comment:
This select statement is a common pattern. Replace with `timed.Wait(ctx, duration)` (timed package is part of go-concert).

Reply:
this does not drain the channel now is designed to be used in select statements so i will skip this

Reply:
agreed offline that further refactoring of the funcs should be performed as a followup resulting in breaking one big function into smaller parts and getting rid of select statments in favor of proposed go-concert where possible

Reply:
++ on follow up refactoring.